public/app/features/dashboard-scene/scene/DashboardScene.tsx

```diff
@@ -533,6 +534,58 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
     store.delete(LS_PANEL_COPY_KEY);
   }
 
+  public removePanel(panel: VizPanel) {
+    const panels: SceneObject[] = [];
+    const key = panel.parent instanceof LibraryVizPanel ? panel.parent.parent?.state.key : panel.parent?.state.key;
+
+    if (!key) {
+      return;
+    }
+
+    let row: SceneGridRow | undefined;
+
+    try {
+      row = sceneGraph.getAncestor(panel, SceneGridRow);
+    } catch {
+      row = undefined;
+    }
+
+    if (row) {
+      row.forEachChild((child: SceneObject) => {
+        if (child.state.key !== key) {
+          panels.push(child);
+        }
+      });
+
+      row.setState({ children: panels });
+
+      const layout = this.state.body;
+
+      if (layout instanceof SceneGridLayout || layout instanceof SceneFlexLayout) {
+        const children: SceneObject[] = layout.state.children;
+        const index = children.findIndex((child) => child.state.key === row?.state.key);
+
+        children.splice(index, 1, row);
+
+        layout.setState({ children });
+      }
```

Comment:
I'm not sure if this is the best way to do it. I would've expected setting the row state to refresh the dashboard and show that the panel was removed since the layout is updated but it doesn't. I looked into `forceRenderChildren` as well but it doesn't work with the layout body, it rerenders only if I send it the entire scene as a param which seems like a lot of unnecessary computation.

Reply:
@mdvictor i think the issue you are facing is related to the fact that the SceneGridRow is actually not rendering it's children, so changing the state of that object will not result in a row re-render

Reply:
think layout.forceRender() should work here instead

Reply:
Thank you folks, much cleaner. That worked!